Assets/Shaders/Caustics.gdshader

```diff
@@ -66,23 +65,36 @@ varying vec3 positionWS;
 varying vec3 positionOS;
 varying vec3 screen_color;
 varying float bbox_mask;
+
 void fragment(){
 	float depth = texture(DEPTH_TEXTURE, SCREEN_UV).x;
-	vec3 ndc = vec3(SCREEN_UV, depth) * 2.0 - 1.0;
+	vec3 ndc;
+	if (OUTPUT_IS_SRGB){
+		 // Compatibility. Uses OpenGL NDC space: [-1,1] for NDC.xy, [-1,1] for NDC.z
+		 ndc = vec3(SCREEN_UV, depth) * 2.0 - 1.0;
+	} else {
+		// Forward+ or Mobile. Uses Vulkan NDC space: [-1,1] for NDC.xy, [0,1] for NDC.z
+		ndc = vec3(SCREEN_UV * 2.0 - 1.0, depth); 
+	}
 	vec4 view = INV_PROJECTION_MATRIX * vec4(ndc, 1.0);
 	view.xyz /= view.w;
 	float linear_depth = -view.z;
-	vec4 world = vertex_camera * INV_PROJECTION_MATRIX * vec4(ndc, 1.0);
+	vec4 world = INV_VIEW_MATRIX * INV_PROJECTION_MATRIX * vec4(ndc, 1.0);
 	
 	positionWS = world.xyz / world.w;
-	positionOS = positionWS - WORLD_MATRIX[3].xyz;
-	bbox_mask = check_encompass_point(positionWS);
+	positionOS = positionWS - MODEL_MATRIX[3].xyz;
+	bbox_mask = check_encompass_point(positionWS, MODEL_MATRIX);
 	
 	screen_color = texture(SCREEN_TEXTURE, SCREEN_UV).rgb;
 	
-	// Want to do all the work in the light function
-	// So we use additive blending and set albedo to zero
-	ALBEDO = vec3(0.0);
+	// The caustics are processed in the light() function, so we can support 
+	// multiple lights. Set the albedo to 1.0 and use additive blending.
+	ALBEDO = vec3(1.0);
+	
+	// These are the implicit default values, anyway.
+	METALLIC = 0.0;
```

Comment:
These may be unnecessary, but I like the explicit definition since defaults can change between versions.